common/lib/common-utils/.eslintrc.js

```diff
@@ -5,16 +5,55 @@
 
 module.exports = {
     "extends": [
-        "@fluidframework/eslint-config-fluid"
+        "@fluidframework/eslint-config-fluid/recommended"
     ],
     "parserOptions": {
-        "project": [ "./tsconfig.json", "./src/test/mocha/tsconfig.json", "./src/test/jest/tsconfig.json", "./src/test/types/tsconfig.json" ]
+        "project": [
+            "./tsconfig.json",
+            "./src/test/mocha/tsconfig.json",
+            "./src/test/jest/tsconfig.json",
+            "./src/test/types/tsconfig.json",
+        ]
     },
     "rules": {
         "@typescript-eslint/no-non-null-assertion": "off",
         "@typescript-eslint/strict-boolean-expressions": "off",
+        "@typescript-eslint/no-implied-eval": "error",
+        "@typescript-eslint/promise-function-async": "error",
+        "default-case": "error",
         "no-bitwise": "off",
         "no-null/no-null": "off",
-        "prefer-rest-params": "off"
-    }
+        "prefer-rest-params": "off",
+        "import/no-internal-modules": "error",
+        "import/no-nodejs-modules": [
+            "error", {
+                "allow": [
+                    "node:assert",
+                    "node:events",
+                    "node:fs",
+                    "node:http",
+                    "node:path",
+                    "node:perf_hooks",
+                    "node:process",
+                ]
+            }
+        ],
+    },
+    "overrides": [
+        {
+            // Rules only for test files
+            "files": [
+                "**/*.spec.ts",
+                "src/test/**"
+            ],
+            "rules": {
+                "@typescript-eslint/explicit-function-return-type": "off",
+                "unicorn/consistent-function-scoping": [
+                    "error", {
+                        "checkArrowFunctions": false
+                    }
+                ],
+            },
+        }
```

Comment:
Ignore this file -- it's just used to test rule changes on the local project more easily. Once this PR is reviewed, I'll move the appropriate rules to the shared config.